SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -89,23 +93,20 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	//checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
+	// Bind the ProcessServerTravel delegate to the spatial variant. This ensures that if ServerTravel is called and Spatial networking is enabled, we can travel properly.
+	LoadedWorld->SpatialProcessServerTravelDelegate.BindStatic(SpatialProcessServerTravel);
+
+	// Handle Spatial connection configurations.
+	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
 
 	// Set the timer manager.
 	TimerManager = &LoadedWorld->GetTimerManager();
 
 	// Set up manager objects.
 	EntityRegistry = NewObject<UEntityRegistry>(this);
 
-	// Handle Spatial connection configurations.
-	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
-
-	if (Connection != nullptr)
-	{
-		return;
-	}
-
-	Connection = NewObject<USpatialWorkerConnection>();
+	// Grab the SpatialWorkerConnection from the SpatialGameInstance (stored there for persistence in server travel). 
+	Connection = Cast<USpatialGameInstance>(GetWorld()->GetGameInstance())->SpatialConnection;
```

Comment:
In an unmodified GDK we will indeed always have `USpatialGameInstance` here. However, please assert that `GetGameInstance()->IsA(USpatialGameInstance::StaticClass())`. This is to help people making custom changes to the GDK.

Reply:
But if it's a child of `USpatialGameInstance`, won't `Cast<USpatialGameInstance>` work just fine?

Reply:
`Cast<>` already returns null if it's not a castable class, so it'll just crash here in that case.

Reply:
I guess could we do a Cast and `checkf` that it's not null?

Reply:
Discussed offline, going with how it's currently implemented since the null return should be enough info for users.